justin-tan/week2/fri/ga-bank/js/atm.js

```diff
@@ -0,0 +1,220 @@
+    }
+}
+
+//===========================================================
+
+
+//function for CHECKING BALANCE background color
+const backgroundCheckingColor = function () {
+    if ( ($checkingBalance === 0 || $(`#checking-balance`).text() === `Insufficient $$`) ) {
+        $('#checking-balance').css('background-color', 'red');
+    } else {
+        $('#checking-balance').css('background-color', 'rgb(227, 227, 227');
+    }
+}
+
+//===========================================================
+
+
+//event handlers for checking account deposit and withdrawal buttons
+$('#checking-deposit').on( 'click', checkingDeposit );
+$('#checking-withdraw').on( 'click', checkingWithdrawal );
+
+
+//event handlers for checking account background color
+$('#checking-withdraw').on( 'click', backgroundCheckingColor );
+$('#checking-deposit').on( 'click', backgroundCheckingColor );
+
+
+//=========================================================
+//=========================================================
+//=========================================================
+
+
+//function for SAVINGS deposit
+const savingsDeposit = function() {
+    //set input value into a number
+    const $savingsValue = $('#savings-amount').val();
+    const $savingsAmount = parseInt($savingsValue);
+
+    //add input amount into account balance
+    $savingsBalance = $savingsBalance + $savingsAmount;
+    $('#savings-balance').text(`$${$savingsBalance}`);
+}
+
+//===========================================================
+
+
+//function for SAVINGS withdrawal
+const savingsWithdrawal = function() {
+    //set variable for total balance
+    let totalBalance = $checkingBalance + $savingsBalance
+
+    //set input value into a number
+    const $savingsValue = $('#savings-amount').val();
+    const $savingsAmount = parseInt($savingsValue);
+    
+    //set conditions withdrawal:
+    //when you have sufficient balance to initiate withdraw
+    if ( $savingsBalance >= $savingsAmount ) {
+        $savingsBalance = $savingsBalance - $savingsAmount;
+        $('#savings-balance').text(`$${$savingsBalance}`);
+    
+    //when overdraft protection initiated
+    } else if ( $savingsBalance < $savingsAmount && $savingsAmount <= totalBalance) {
+        
+        //set variable for overdraft
+        let overdraft = $savingsBalance - $savingsAmount;
+
+        //formula for remaining balances for both account
+        $savingsBalance = $savingsBalance - $savingsBalance;
+        $checkingBalance = $checkingBalance + overdraft;
+        $('#checking-balance').text(`$${$checkingBalance}`) ;
+        $('#savings-balance').text(`$${$savingsBalance}`);
+
+    //when you have insufficient total balance to withdraw
+    } else if ( $savingsAmount > totalBalance ) {
+        $('#savings-balance').text(`Insufficient $$`);
+    }
+}
+
+//===========================================================
+
+//function for SAVINGS BALANCE background color
+const backgroundSavingsColor = function () {
```

Comment:
you already have a `backgroundCheckingColor` that does the same thing
you can refactor that to take in a parameter, being the account balance name

```
const backgroundBalance = function (accBalance) { // `#checking-balance` or  #savings-balance'
    if ( ($checkingBalance === 0 || $(accBalance).text() === `Insufficient $$`) || ($savingBalance === 0)) {
        $(accBalance).css('background-color', 'red');
    } else {
        $(accBalance).css('background-color', 'rgb(227, 227, 227');
    }
}


$('#checking-withdraw').on( 'click', backgroundBalance('#checking-balance');


```